libraries/chain/finality/qc.cpp

```diff
@@ -309,10 +375,54 @@ std::optional<qc_t> aggregating_qc_t::get_best_qc(block_num_type block_num) cons
    return std::optional<qc_t>{qc_t{block_num, std::move(*active_best_qc), {}}};
 }
 
+// A dual finalizer votes on both active and pending finalizer policies
+void aggregating_qc_t::verify_dual_finalizers_votes(const qc_t& qc) const {
+   auto build_vote_index = [&](const std::vector<finalizer_authority> finalizers, std::set<bls_public_key>& keys, std::map<bls_public_key, uint32_t>& index) {
```

Comment:
`finalizers` should be `const&`

Reply:
https://github.com/AntelopeIO/spring/pull/475/commits/58edf7588ed6bafa7a709d0e8c866d6d33f803e6